core/sdk-core/src/main/java/software/amazon/awssdk/core/progress/RequestCycleEventType.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.progress;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Request life cycle events
+ */
+@SdkPublicApi
+public enum RequestCycleEventType implements ProgressEventType {
```

Comment:
Should we document what information the customer can expect with each event type?